cache/in-memory/src/lib.rs

```diff
@@ -116,7 +116,7 @@ struct InMemoryCacheRef {
     presences: DashMap<(Option<GuildId>, UserId), Arc<CachedPresence>>,
     roles: DashMap<RoleId, GuildItem<Role>>,
     unavailable_guilds: DashSet<GuildId>,
-    users: DashMap<UserId, Arc<User>>,
+    users: DashMap<UserId, (Arc<User>, HashSet<GuildId>)>,
```

Comment:
This may be one of the cases where a `BTreeSet` is better as it has a lower boud on memory consumption. This is afterall a structure that will be made a ton of.

Reply:
Sounds good, I've switched out `HashSet` for `BtreeSet`